packages/plugin-connector/src/data-sources/schemas/query-runner.schema.js

```diff
@@ -1,15 +1,90 @@
 import { z } from 'zod';
 
-export const QueryResultSchema = z.object({
-	rows: z.array(z.record(z.string(), z.any())),
-	columnTypes: z.array(
-		z.object({
-			name: z.string(),
-			evidenceType: z.enum(['boolean', 'number', 'string', 'date']),
-			typeFidelity: z.union([z.literal('precise'), z.literal('inferred')])
-		})
+export const QueryResultSchema = z
+	.object({
+		// Note that this only validates that the first item in the array
+		// is a record with string keys. If the connector returns some
+		// inconsistent array (e.g. [{}, 1]), it will not detect the
+		// invalid row.
+		rows: z.array(z.any()).refine(
+			(data) => {
+				// result is not an array, fail
+				if (!Array.isArray(data)) return false;
```

Comment:
Another way would be to have the `z.any()...` as another `zod` object..

e.g.

```js

rows: z.array(rowItem)

//...

export const rowItemModal = z.record(z.string), z.any())
export type rowItem = z.infer<typeof rowItemModal>;
```

Reply:
I tried that initially and that didn't prevent it from validating every row, doing `z.any()` and then checking the first row in the refine method was the only thing that prevented the full scan